### DISCUSSION TITLE
How to add `manager` to People result

### DISCUSSION THREAD

**Discussion by simesy** [microsoft-search/pnp-modern-search#2069] [Q&A] [answered]
I want to build a People Directory that includes Manager information. I can do everything with a custom template and I feel good about it. I have played with SharePoint Search and Microsoft Search and it should be possible to use either.

My only issue is displaying the manager information. Does anyone know how to add `manager` to the search result (the `userPrincipalName` and `displayName`. Is this something that can be done just through the webpart configuration? The single graph call is `https://graph.microsoft.com/v1.0/contacts/{id}/manager`.

**Comment by wobba** [comment]
The crawled property `People:Manager` is not mapped to a managed property by default. You have to map it yourself (and re-index profiles) in order to return it back in the search result.

**Comment by cluber22** [comment]
> I want to build a People Directory that includes Manager information. I can do everything with a custom template and I feel good about it. I have played with SharePoint Search and Microsoft Search and it should be possible to use either.
> 
> My only issue is displaying the manager information. Does anyone know how to add `manager` to the search result (the `userPrincipalName` and `displayName`. Is this something that can be done just through the webpart configuration? The single graph call is `https://graph.microsoft.com/v1.0/contacts/{id}/manager`.

@simesy Did you ever sort this out?

@wobba  Thanks for that however I unsuccessfully managed to get the manager data using Microsoft Search, but it does state on the documentation MS Search is not for production use and to use SharePoint Search. https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/#microsoft-search

On the SharePoint Search side I managed to map the fields however it was null for some people, even though I can see the manager entered on the SharePoint Admin Manage User Profiles page for the same users.

![sharepoint-search-managed-prop](https://github.com/user-attachments/assets/c5c201c3-8615-4757-9eb3-19f6eef49315)


Any help would be greatly appreciated.

**Reply by kasperbolarsen** [reply]
Did you reindex the People search index? This script is the most updated I know of: https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html?tabs=pnpps

**Reply by cluber22** [reply]
@kasperbolarsen Thank you for this, I will give the script a go. I obviously don't fully understand the SharePoint search mechanisms as much as I thought! So essentially it doesn't reindex everything every night, it will only look at recently updated profiles and index those. Good to know.

**Reply by kasperbolarsen** [reply]
Indeed,  back in the On Premises days we could set it up to do a full crawl at a given interval,  but in Online it is MS that is controlling the indexer.

**Reply by cluber22** [reply]
@kasperbolarsen That re-index worked and it took about 24 hours.

The problem now is that field contains data in the following format.

`i:0#.f|membership|forename.surname@domain.com`

How do I get the manager's display name?

**Reply by PFudala** [reply]
> How do I get the manager's display name?

Hi @cluber22,
Did you find a solution to this ?

**Reply by cluber22** [reply]
> > How do I get the manager's display name?
> 
> Hi @cluber22, Did you find a solution to this ?

Unfortunately not, although I've not spent a great deal of time on it. I tried a few things which didn't work.

**Reply by PFudala** [reply, marked as answer]
> Unfortunately not, although I've not spent a great deal of time on it. I tried a few things which didn't work.

I finally found a solution, using MGT option (page 2 from WebPart properties) : 

![image](https://github.com/user-attachments/assets/c34ed89b-f270-4033-8063-8178c011df19)

Then, in my custom template, I used a custom manaed property "IndexedManager" from Search Schema : 

![image](https://github.com/user-attachments/assets/b6eec55a-1b5f-4f8a-9b0e-e288b36b4758)

```
<div class="contact-row">
    <span class="contact-row-title">Manager : </span>
    <mgt-person person-query="{{getUserEmail (last (split IndexedManager '|'))}}" view="oneline"></mgt-person>
    <div class="contact-row-action">
        <a href="mailto:{{getUserEmail (last (split IndexedManager '|'))}}"><pnp-icon data-name="Mail" aria-hidden="true"></pnp-icon></a>
    </div>
</div>
```
To obtain the following result : 

![image](https://github.com/user-attachments/assets/dd80ce0a-5407-4165-9824-c7918a9b55d1)

Hope it helps 🙂

**Reply by kasperbolarsen** [reply]
Nice solution :-) and most likely it will work with the Assistant and Dotted-line manager as well

**Reply by kasperbolarsen** [reply]
@PFudala , what would the correct French term for "Use Microsoft Graph Toolkit" be? I'll update the translation files if you can provide the correct term

![image](https://github.com/user-attachments/assets/3b4c0dad-3dad-4105-818d-f9827abaff55)

**Reply by PFudala** [reply]
> @PFudala , what would the correct French term for "Use Microsoft Graph Toolkit" be? I'll update the translation files if you can provide the correct term

Simply "Utiliser Microsoft Graph Toolkit" would be nice 👍
